.pipeline/lib/deploy.js

```diff
@@ -1,42 +1,45 @@
 'use strict';
 
-const { OpenShiftClientX } = require('pipeline-cli');
+const { OpenShiftClientX } = require('@bcgov/pipeline-cli');
 const path = require('path');
 
-module.exports = () => {
-  const oc = new OpenShiftClientX();
-  oc.globalArgs.namespace = `devhub-${oc.options.env}`;
-  const templateFile = path.resolve(__dirname, '../../openshift/dc.yaml');
-  const appName = 'reggie-api';
-  const buildNamespace = 'devhub-tools';
-  const buildVersion = `build-v${oc.options.pr}`;
-  const deploymentVersion = `${oc.options.env}-1.0.0`;
-  // remove pr in prefix for test and prod environemnt:
-  const projectPrefix =
-    oc.options.env === 'dev' ? `-${oc.options.env}-${oc.options.pr}` : `-${oc.options.env}`;
+module.exports = (settings)=>{
+  const phases = settings.phases;
+  const options = settings.options;
+  const phase = options.env;
+  const changeId = phases[phase].changeId;
+  const oc = new OpenShiftClientX(Object.assign({ namespace: phases[phase].namespace }, options));
+  const templatesLocalBaseUrl = oc.toFileUrl(path.resolve(__dirname, '../../openshift'));
+  let objects = [];
 
   // set the rest of the env vars:
   const extraParams = {
     RM_HOST_VALUE: 'https://repo-mountie-devhub-prod.pathfinder.gov.bc.ca/bot/github/membership',
```

Comment:
i wonder if this would be best placed in an outside config file that is imported in. I'm not doing it my self but seems like a good idea..

Reply:
yes going to move that to a configmap/secret

Reply:
btw I'm still wip on the pipeline in this pr. if it gets too big, I'll split it.

Reply:
its mostly boiler plate bcdk stuff so i  think its fine for now :)